Misc Updates

tgraf opened the pull request.

mchalla commented on bpf/bpf_lxc.c:
```diff
@@ -595,10 +598,13 @@ static inline int __inline__ ipv4_policy(struct __sk_buff *skb, int ifindex, __u
+	 * passed through the allowed consumer. */
+	verdict = policy_can_access(&POLICY_MAP, skb, src_label);
+	if (unlikely(ret == CT_NEW)) {
+		if (verdict != TC_ACT_OK)
```
Is there a reason to move the verdict check inside the if(ret == CT_NEW) block ?

tgraf replied:
It leads to a better fast path because `CT_NEW` should only be seen for the first packet of a flow. We only need to look at the policy verdict for new flows. All packets have to go through the policy table though for accounting. We can still optimize this by making the account part optional and it that case only perform the map lookup when the flow is first created.

mchalla replied:
Looks good to me. thanks.

aanm commented on daemon/daemon/docker_watcher.go:
```diff
@@ -305,19 +305,19 @@ func (d *Daemon) updateContainer(container *types.Container, isNewContainer bool
 	if try >= maxTries {
 		if container.IsDockerOrInfracontainer() {
-			return fmt.Errorf("Unable to store security context %d for container %s", secCtxlabels.ID, dockerID)
+			return fmt.Errorf("No manage request in time, container %s is likely managed by other networking plugin.", dockerID)
```
Is likely, but is also likely the event was lost, right?

tgraf replied:
The timeout should be long enough for a polling cycle, right?

aanm replied:
We are polling the docker service every 30 seconds, so if we don't catch the event now, we'll catch it later.

aanm commented on examples/docker-compose/README.md:
```diff
@@ -5,13 +5,30 @@ docker containers using the cilium docker image.
+
+If you want to install the dependencies manually, you need:
  - docker-compose (>=1.7.1)
- - Linux-kernel (>=4.8.0)
```
Should this file be updated as well? https://github.com/cilium/cilium/blame/master/README.md#L95

tgraf replied:
The commit already does.

aanm replied:
You're right, miss it.

tgraf replied:
Thanks for the review

tgraf committed:
AUTHORS: Add Alban Crequy

Acked-by: André Martins <andre@cilium.io>
Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
packer: Bump version for 0.1.2 build

Acked-by: André Martins <andre@cilium.io>
Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
daemon: Better log messages around request loop

Improves the log messages around the process of waiting on the orchestration
system to request management of the networking pieces.

Acked-by: André Martins <andre@cilium.io>
Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
doc: Provide better documentation on kernel and iproute2 requirement

Reported-by: Salvatore Orlando
Acked-by: André Martins <andre@cilium.io>
Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
bpf: Fix typo in skb_change_tail probe invocation

Acked-by: André Martins <andre@cilium.io>
Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
bpf: Cleanup policy skip mark

Acked-by: Madhu Challa <madhu@cilium.io>
Signed-off-by: Thomas Graf <thomas@cilium.io>
